### ISSUE TITLE
python wrapper : Question about online learning , update option parameters and predict_probability  

### ISSUE THREAD

**Issue by vatn** [JohnLangford/vowpal_wabbit#1288]
Hello, I'm trying to learning vowpal_wabbit using python wrapper.
I now try to test with titanic dataset from kaggle. After reading the document, I still have some questions: 

1) online learning
Does "fit" function means update the model?
for example: 
`
model_test = VWClassifier(passes=1,random_seed = 0,loss_function = 'logistic',link = 'logistic' )
model_test.fit(X_train, y_train)
model_test.fit(X_test, y_test)
`
Here, does it means that I fit the model_test  with X_train,y_train , then the fitted model_test is updated with (X_test, y_test)?

2) update option parameters:
I didn't find the parameters in sklearn interface(VWClassifier) which correspond to these command line arguments:  --adaptive, --normalized and --invariant.
So, if I define a model by              
`model_test = VWClassifier(passes=1,random_seed = 0,loss_function = 'logistic',link = 'logistic' )`
Does it mean all these three options are "on"  by default ?    
And power_t, initial_t and learning rate is for which kind of adaptive update rules here?
If  'ftrl' or 'bfgs' is given True, does it means  the three options "--adaptive, --normalized and --invariant." will close?

3) predict probability

When I add the link = "logistic" to the model, the decision_function will output the probability.
However, when I use the "predict" function to predict class labels, perhaps there is an error in the output predicted class.(because the "predict" function always use the threshold=0 to make predictions ).
So, perhaps it is interesting to change threshold to 0.5, when link = 'logistic' ?

And also, may I know the formula to transform the raw predictions to probability?(More precisely, I mean the relationship like sigmoid between the predictions output by decision_function when the model is without link = 'logistic' and the probability output by decision_function when the model is with link = 'logistic')

Thanks a lot!

**Comment by JohnLangford** [comment]
1) I believe "yes"
2) Yes, these are the defaults.   Adaptive is adagrad.  FTRL and BFGS are different base learning algorithms which take different parameters.
3) You can control the link function in general with --link.  The default is identity and squared loss although the sklearn interface defaults to logistic link & loss.

**Comment by vatn** [comment]
I understand better now, 
Thanks so much for your answer!

**Comment by mlisovyi** [comment]
May I clarify the third question: it seems that the proper way to use `link='logistic'` with the sklearn interface is to also add `pos_threshold=0.5` as an input parameter (to be picked as an argument by `ThresholdingLinearClassifierMixin`). Is that correct? If so, one should clarify the note `note - don't try to apply link='logistic' on top of the existing functionality` [here on the code](https://github.com/JohnLangford/vowpal_wabbit/blob/master/python/vowpalwabbit/sklearn_vw.py#L499) to avoid confusion
